pkgs/tools/networking/slack-cli/wrapper.nix

```diff
@@ -12,4 +12,10 @@ writeShellScriptBin "slack" ''
 
   export PATH=${lib.makeBinPath [ curl jq ]}:"$PATH"
   exec ${slack-cli}/bin/slack "$@"
-''
+'') // {
```

Comment:
Personally I'd just inline the wrapper into the main package. Any reason you chose this approach?

Reply:
There is indeed! See https://github.com/NixOS/nixpkgs/pull/44102, but the tl;dr is that inlining the wrapper into the main package breaks `overrideAttrs`, so as a user of the package you can't eg add patches.

Reply:
It doesn't need to be 2 packages at all: Just "insert" the wrapper into the original derivation like this: `ln -s ${wrapper}/bin/slack-cli $out/bin/slack-cli`, while moving the original slack-cli binary to `$out/bin/.slack-cli-wrapped_`

Reply:
Oh, I see what you're saying! I'll do that.